packages/material-ui-csskit/src/responsiveTypography.js

```diff
@@ -0,0 +1,88 @@
+import convertLength from 'convert-css-length';
+import responsiveProperty from './responsiveProperty';
+import alignProperty from './alignProperty';
+import fontGrid from './fontGrid';
+
+function isUnitless(value) {
+  return String(parseFloat(value)).length === String(value).length;
+}
+
+/**
+ *
+ * responsiveTypography params
+ * @typedef {Object} responsiveParams
+ * @property {number} maxScale - The scale that applies to the highest breapoint
+ * @property {Object} breakpointSettings - the breakpoints settings in the material-ui theme
+ * @property {Array.string} breakpoints - Array of breakpoints (identifiers)
+ * @property {boolean} align - Whether font sizes change slightly so line
+ *     heights are preserved and align to Material Design's 4px line height grid
+ *
+ * @param {Object} typography - the typography settings in the material-ui theme
+ * @param {responsiveParams}
+ * @returns new theme with responsive Typography styles
+ */
+export default function responsiveTypography(typography, options = {}) {
+  const {
+    maxScale,
+    breakpointSettings,
+    breakpoints = ['sm', 'md', 'lg', 'xl'],
+    align = true,
+  } = options;
+  // Convert between css lengths e.g. em->px or px->rem
+  // Set the baseFontSize for your project. Defaults to 16px (also the browser default).
+  const convert = convertLength(typography.htmlFontSize);
+  const output = { ...typography };
+  const breakpointValues = breakpoints.map(x => breakpointSettings.values[x]);
+
+  [
+    'h1',
+    'h2',
+    'h3',
+    'h4',
+    'h5',
+    'h6',
+    'subtitle1',
+    'subtitle2',
+    'body1',
+    'body2',
+    'caption',
+    'button',
+    'overline',
+  ].forEach(variant => {
+    const style = output[variant];
+
+    const remFontSize = convert(style.fontSize, 'rem');
+    const minFontSize = parseFloat(remFontSize);
+    const maxFontSize = Math.round(minFontSize * maxScale * 10) / 10;
+    let { lineHeight } = style;
+
+    if (!isUnitless(lineHeight)) {
+      // make it unitless
+      lineHeight = convert(lineHeight, 'rem') / remFontSize;
```

Comment:
I don't think that it works: `string / string => NaN`. It would be great to add a test case for it :).

Reply:
Ups forgot a parsing, will add tests